dashboard/src/components/AppView/AppView.tsx

```diff
@@ -147,23 +158,32 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
   public handleEvent(e: MessageEvent) {
     const msg = JSON.parse(e.data);
     const resource: IResource = msg.object;
-    const key = `${resource.kind}/${resource.metadata.name}`;
+    const item = {
+      [resource.metadata.name]: { item: resource, isFetching: false },
+    };
     switch (resource.kind) {
       case "Deployment":
-        this.setState({ deployments: { ...this.state.deployments, [key]: resource } });
+        this.setState({
+          deployments: { ...this.state.deployments, ...item },
+        });
         break;
       case "Service":
-        this.setState({ services: { ...this.state.services, [key]: resource } });
+        this.setState({
+          services: { ...this.state.services, ...item },
+        });
         break;
       case "Ingress":
-        this.setState({ ingresses: { ...this.state.ingresses, [key]: resource } });
+        this.setState({
+          ingresses: { ...this.state.ingresses, ...item },
+        });
         break;
     }
   }
 
-  public get isLoading(): boolean {
+  // isAppLoading checks if the given app has been collected from tiller
+  public get isAppLoading(): boolean {
```

Comment:
why do we need a getter instead of a normal function? I'm just curious mostly, but I also think explicit functions are better because I now can't tell if this is a property or a function when I see it being used. Why couldn't appInfo be a getter?

Reply:
this was a getter before and I didn't change it. I don't have a good reason why this is not a function. Actually I don't think this function is useful anyway.